tox.ini

```diff
@@ -82,15 +82,22 @@ deps =
 
     mock
     lxml
+    coverage
+
 commands =
     /usr/bin/find "{toxinidir}" -name '*.pyc' -delete
     # Make all greater-thans equals so we test the oldest version of our direct
     # dependencies, but make the pyopenssl 17.0, which can work against an
     # OpenSSL 1.1 compiled cryptography (as older ones don't compile on Travis).
     /bin/sh -c 'python -m synapse.python_dependencies | sed -e "s/>=/==/g" -e "s/psycopg2==2.6//" -e "s/pyopenssl==16.0.0/pyopenssl==17.0.0/" | xargs pip install'
-    # Install Synapse itself. This won't update any libraries.
+
+    # Add this so that coverage will run on subprocesses
+    /bin/sh -c 'echo "import coverage; coverage.process_startup()" > {envsitepackagesdir}/../sitecustomize.py'
+
     pip install -e .
-    {envbindir}/trial {env:TRIAL_FLAGS:} {posargs:tests} {env:TOXSUFFIX:}
+
+    {envbindir}/coverage run "{envbindir}/trial"  {env:TRIAL_FLAGS:} {posargs:tests} {env:TOXSUFFIX:}
```

Comment:
do we actually need to run the coverage checks here? is there any reason to imagine the coverage will be any different to the other runs?

Reply:
The same could be said for a lot of tests tbh, I'd rather we just consistently did coverage stats than have to figure out which tests do and which don't.

Reply:
fair